pkg/edgehub/config/config_test.go

```diff
@@ -0,0 +1,508 @@
+		CertFilePath: "/tmp/edge.crt",
+		KeyFilePath:  "/tmp/edge.key",
+	}
+	edgeHubConfigYaml2 := edgeHubConfigYaml{
+		webSocketConfigYaml2,
+		controllerConfigYaml,
+	}
+
+	webSocketConfigYaml3 := webSocketConfigYaml{
+		CertFilePath:     "/tmp/edge.crt",
+		KeyFilePath:      "/tmp/edge.key",
+		HandshakeTimeout: "500",
+		WriteDeadline:    "100",
+		ReadDeadline:     "100",
+	}
+	edgeHubConfigYaml3 := edgeHubConfigYaml{
+		webSocketConfigYaml3,
+		controllerConfigYaml,
+	}
+
+	webSocketConfigYaml4 := webSocketConfigYaml{
+		Url:              "ws://127.0.0.1:20000/fake_group_id/events",
+		KeyFilePath:      "/tmp/edge.key",
+		HandshakeTimeout: "500",
+		WriteDeadline:    "100",
+		ReadDeadline:     "100",
+	}
+	edgeHubConfigYaml4 := edgeHubConfigYaml{
+		webSocketConfigYaml4,
+		controllerConfigYaml,
+	}
+
+	webSocketConfigYaml5 := webSocketConfigYaml{
+		Url:              "ws://127.0.0.1:20000/fake_group_id/events",
+		CertFilePath:     "/tmp/edge.crt",
+		HandshakeTimeout: "500",
+		WriteDeadline:    "100",
+		ReadDeadline:     "100",
+	}
+	edgeHubConfigYaml5 := edgeHubConfigYaml{
+		webSocketConfigYaml5,
+		controllerConfigYaml,
+	}
+	//Testcases
+	tests := []struct {
+		name    string
+		test    testYamlGenerator
+		wantErr bool
+	}{
+		//Positive Testcase with all values correctly specified
+		{"Test_getWebSocketConfig1: Proper Input",
+			testYamlGenerator{
+				edgeHubConfigYaml1,
+			},
+			false},
+		//Positive Testcase which uses default input for Handshake-timeout, Write Dealine & Read Deadline
+		{"Test_getWebSocketConfig2: Use Default input for Handshake-timeout, Write Dealine & Read Deadline ",
+			testYamlGenerator{
+				edgeHubConfigYaml2,
+			},
+			false},
+		//Positive Testcase with no URL specified
+		{"Test_getWebSocketConfig3: No URL provided ",
+			testYamlGenerator{
+				edgeHubConfigYaml3,
+			},
+			false},
+		//Negative  Testcase with no CertFile  specified
+		{"Test_getWebSocketConfig4: No Cert File provided ",
+			testYamlGenerator{
+				edgeHubConfigYaml4,
+			},
+			true},
+		//Negative  Testcase with no KeyFile  specified
+		{"Test_getWebSocketConfig5: No Key File provided",
+			testYamlGenerator{
+				edgeHubConfigYaml5,
+			},
+			true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			generateTestYaml(tt.test)
+			err := util.LoadConfigFromLocation("/tmp/kubeedge/testData")
+			if err != nil {
+				t.Error("Unable to load the configuration file")
+			}
+			removeTemporaryDirectory("/tmp/kubeedge")
+			if err := getWebSocketConfig(); (err != nil) != tt.wantErr {
+				t.Errorf("getWebSocketConfig() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+//This function loads the testing config file  tests whether the controller  configurations are loaded correctly with the values from the config file
+func Test_getControllerConfig(t *testing.T) {
+	webSocketConfigYaml := webSocketConfigYaml{}
+	controllerConfigYaml1 := controllerConfigYaml{
```

Comment:
i feel instead of creating different structures here, we can directly create them in tests := []struct. @m1093782566 @rohitsardesai83 Thoughts ?

Reply:
+1